ocrd/ocrd/task_sequence.py

```diff
@@ -38,14 +39,18 @@ def __init__(self, executable, input_file_grps, output_file_grps, parameter_path
         self.parameter_path = parameter_path
 
     def validate(self):
-        if self.parameter_path and not os.access(self.parameter_path, os.R_OK):
-            raise Exception("Parameter file not readable: %s" % self.parameter_path)
-        if not self.input_file_grps:
-            raise Exception("Task must have input file group")
-        if not self.output_file_grps:
-            raise Exception("Task must have output file group")
         if not which(self.executable):
             raise Exception("Executable not found in PATH: %s" % self.executable)
+        result = run([self.executable, '--dump-json'], stdout=PIPE, check=True, universal_newlines=True)
+        ocrd_tool_json = json.loads(result.stdout)
+        # TODO check for required parameters in ocrd_tool
+        if self.parameter_path:
+            parse_json_string_or_file(self.parameter_path)
```

Comment:
Yes, this finally opens the possibility of also checking that:
- the parameter file does not contain keys that are not in tool.json
- the parameter file specifies a value for every parameter that does not have a default in tool.json (#246)

Reply:
I'll add these in subsequent PRs, #251 #246 are not forgotten.